From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Toffikk <slodkitofik@gmail.com>
Date: Thu, 24 Jul 2025 21:37:59 +0200
Subject: [PATCH] feature panic


diff --git a/net/minecraft/CrashReport.java b/net/minecraft/CrashReport.java
index a9281eb0a66db867f8e9eab4742cfa54867837ab..16e0a6185c6124d47c30d26b9c1d0bc736d141db 100644
--- a/net/minecraft/CrashReport.java
+++ b/net/minecraft/CrashReport.java
@@ -233,6 +233,6 @@ public class CrashReport {
 
     public static void preload() {
         //MemoryReserve.allocate(); // Paper - Disable memory reserve allocating
-        new CrashReport("Don't panic!", new Throwable()).getFriendlyReport(ReportType.CRASH);
+        new CrashReport("Do panic!", new Throwable()).getFriendlyReport(ReportType.CRASH); // Baguette - panic.
     }
 }
